rule-002/src/services/application.service.ts

```diff
@@ -0,0 +1,101 @@
+import axios from 'axios';
+import * as Koa from 'koa';
+import { TransactionCheck } from '../classes/transaction-check';
+import { IRequest, IResponse, ITypologies } from '../interfaces/iRule002';
+import { ArangoDBService } from './arango-client.service';
+import { LoggerService } from './logger.service';
+
+const arangodb = new ArangoDBService();
+
+export class ApplicationService {
+  async main(ctx: Koa.Context): Promise<void> {
+    LoggerService.log('Received Online Request - Status: Online');
+    ctx.status = 200;
+    ctx.body = 'frm-rule-002.';
+  }
+
+  async getOnline(ctx: Koa.Context): Promise<void> {
+    LoggerService.log('Received Online Request - Status: Online');
+    ctx.status = 200;
+    ctx.body = 'frm-rule-002 is online.';
+  }
+
+  async call(ctx: Koa.Context): Promise<void> {
+    LoggerService.log('Start - Handle execute request');
+
+    try {
+      const request: IRequest = {
+        transaction: new TransactionCheck(ctx.request.body.transaction),
+      };
+
+      const result: IResponse = {
+        transaction: request.transaction,
+      };
+
+      const data = ctx.request.body.typologies as ITypologies;
+
+      LoggerService.log(`TransactionID ${request.transaction.TransactionID}`);
+
+      // Get the transaction object include payer and payee
+      const transactionInfoQuery = `
+        FOR doc IN transactions
+          FILTER doc._id == "${request.transaction.TransactionID}"
+          RETURN doc
+          `;
+
+      const transactionData = await arangodb.query(transactionInfoQuery);
+
+      if (transactionData && transactionData[0].length > 0) {
+        const payeeId = transactionData[0][0]._to;
+
+        const payeeTransactionsQuery = `
+          FOR v, e, p IN 2..2 OUTBOUND
+            "${payeeId}"
+            GRAPH "transactions"
+            FILTER e._to == "${payeeId}"
+            return p
+          `;
```

Comment:
I think this is a possible injection vulnerability. Need to check Swagger in TMS.